met-api/src/met_api/resources/survey.py

```diff
@@ -132,3 +133,28 @@ def put():
             return ActionResult.error(str(err))
         except ValueError as err:
             return ActionResult.error(str(err))
+
+
+@cors_preflight('PUT,OPTIONS')
+@API.route('/<survey_id>/link/engagement/<engagement_id>')
+class SurveyLink(Resource):
```

Comment:
hm... generally its just a put on survey/:id ; we dont need the link action in the rest call ... 
but we can treat this as an exception since linking is a major thing...
normally we dont need to write multiple rest calls for each action in the UI..
call put on the suveys/:id and see if the request json has engId ;if so update it...

Reply:
That was my initial way. However, when I implemented that and allowed for engagement_id to be updated in the model class for the update method. I had to put checks for when updating the survey (say adding components) to not update engagement_id with None and lose the link. When I saw that, I felt uncomfortable using the update method to link and to update survey components. I felt link should be on its own.

Reply:
hm..since linking is major thing for us ,its okay to have like this..but if its a smaller operation , we should try to accomdate in the Put itself..
yeah..points agreed..we have to validate the None and all..More code for sure